### PR TITLE
add support for metrics in lookout ingester

### PR DESCRIPTION
**Pull request by dejanzele**

### CODE REVIEW

File: internal/common/startup.go [hunk 1]
```diff
@@ -125,7 +125,7 @@ func ServeMetricsFor(port uint16, gatherer prometheus.Gatherer) (shutdown func()
 	return ServeHttp(port, mux)
 }
 
-// ServeHttp starts a HTTP server listening on the given port.
+// ServeHttp starts an HTTP server listening on the given port.
```
[the remark this answers is not in the document]
Comment (dejanzele) [reply internal/common/startup.go:128]: Actually @kannon92, this seems to be pretty [opinionated](https://www.pristineword.com/grammar-an-http/#:~:text=It's%20an%20HTTP.,takes%20the%20article%20%22an%22.) :D

**Review by kannon92 — APPROVED** [review]
LGTM

**Review by kannon92 — DISMISSED** [review]

File: internal/eventingester/ingester.go [hunk 2]
```diff
@@ -7,6 +7,8 @@ import (
 	"regexp"
 	"sync"
 
+	"github.com/G-Research/armada/internal/lookoutingester/metrics"
```
Comment (suprjinx) [inline internal/eventingester/ingester.go:10]: maybe we should move shared metrics to pkg/ instead  of having cross-service dependency here?

File: internal/lookoutingester/instructions/instructions.go [hunk 3]
```diff
@@ -55,7 +65,12 @@ func Convert(
 // resulting InstructionSet will contain all events that could be parsed, along with the mesageId of the original message.
 // In the case that no events can be parsed (e.g. the message is not valid protobuf), an empty InstructionSet containing
 // only the messageId will be returned.
-func ConvertMsg(ctx context.Context, msg *pulsarutils.ConsumerMessage, userAnnotationPrefix string, compressor compress.Compressor) *model.InstructionSet {
+func (i *Service) ConvertMsg(
```
Comment (suprjinx) [inline internal/lookoutingester/instructions/instructions.go:68]: "i" is a bit cryptic as a name for service reference

File: internal/eventingester/ingester.go (hunk 2 above)
Comment (dejanzele) [reply internal/eventingester/ingester.go]: actually I removed it, as we have better metrics in pulsar itself

**Review by kannon92 — APPROVED** [review]

**Pull request merged**